src/libexpr/primops/context.cc

```diff
@@ -67,10 +61,51 @@ static void prim_unsafeDiscardOutputDependency(EvalState & state, const PosIdx p
 static RegisterPrimOp primop_unsafeDiscardOutputDependency({
     .name = "__unsafeDiscardOutputDependency",
     .arity = 1,
+    .doc = R"(
+      TODO! Old comment is not good.
+
+      Sometimes we want to pass a derivation path (i.e. `pkg.drvPath`) to a builder without causing the derivation to be built
+      (for instance, in the derivation that builds NARs in nix-push, when doing source-only deployment).
+      This primop marks the string context so that `builtins.derivation` adds the path to `drv.inputSrcs` rather than `drv.inputDrvs`.
+
+      Opposite of [`builtins.addDrvOutputDependencies`](#builtins-addDrvOutputDependencies).
+    )",
     .fun = prim_unsafeDiscardOutputDependency
 });
 
 
+static void prim_addDrvOutputDependencies(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    NixStringContext context;
+    auto s = state.coerceToString(pos, *args[0], context, "while evaluating the argument passed to builtins.addDrvOutputDependencies");
+
+    NixStringContext context2;
+    for (auto && c : context) {
+        if (auto * ptr = std::get_if<NixStringContextElem::Opaque>(&c)) {
+            context2.emplace(NixStringContextElem::DrvDeep {
+                .drvPath = ptr->path
+            });
+        } else {
+            /* Can reuse original item */
+            context2.emplace(std::move(c));
+        }
```

Comment:
We might consider making it less lenient. e.g.

### Failing if there isn't exactly one context element which is a `DrvDeep`

Plus: Exactly what we need it for, most conservative.

Minus: Not idemopotent.

### Also allowing a single Opaque to a drv path (end in `.drv`)

Plus: Minimal change from the above to make idempotent.

Minus: More to specify

Reply:
I would definitely keep it idempotent, so that users can just specify what they want and have it work for both old and new nixpkgs.

Reply:
OK I did option 2 then, restricted and idempotent.